Reusing variable previously defined with walrus operator triggers used-before-def error

**To Reproduce**

```python
x = 3
z = (y := str(x)).index(y.lower())
```

**Expected Behavior**

Should be no errors -- this code does not error using mypy v0.991 or prior, so this seems to be a new issue with v1.0.0.

**Actual Behavior**

```error: Name "y" is used before definition  [used-before-def]```

**Your Environment**

- Mypy version used: 1.0.0
- Mypy command-line flags: none
- Python version used: 3.9

--- a/mypy/traverser.py
+++ b/mypy/traverser.py
@@ -253,9 +253,9 @@ class TraverserVisitor(NodeVisitor[None]):
             o.expr.accept(self)
 
     def visit_call_expr(self, o: CallExpr) -> None:
+        o.callee.accept(self)
         for a in o.args:
             a.accept(self)
-        o.callee.accept(self)
         if o.analyzed:
             o.analyzed.accept(self)
 

--- a/test-data/unit/check-python38.test
+++ b/test-data/unit/check-python38.test
@@ -573,6 +573,14 @@ def foo() -> None:
     [x := x + y for y in [1, 2, 3]]
 [builtins fixtures/dict.pyi]
 
+[case testWalrusUsedBeforeDef]
+# flags: --python-version 3.8
+class C:
+    def f(self, c: 'C') -> None: pass
+
+(x := C()).f(y)  # E: Cannot determine type of "y"  # E: Name "y" is used before definition
+(y := C()).f(y)
+
 [case testOverloadWithPositionalOnlySelf]
 # flags: --python-version 3.8
 from typing import overload, Optional

